### ISSUE TITLE
[SX1262] Not sending packets after updating to v5.7

### ISSUE THREAD

**Issue by timmbogner** [jgromes/RadioLib#694]
**Describe the bug**
Under RadioLib 5.6, the following sketch successfully sends via a Heltec LoRa32 v3 SX1262. It is received successfully via a TTGO OLED LoRa v1 (SX1276) but the RSSI is ~120. When updated to RadioLib 5.7, the sender responds identically, but no packet is received by the TTGO.  I'm using Arduino 2.0.4 in Windows 10.

I found out while making this issue that enabling the RadioLib debug options causes packets to be recieved under RL 5.7. When this occurs the RSSI drops to 30. This is my first time working with the SX1262, do I need to do something with the RF switch?

Sorry if this is something I've overlooked!

**To Reproduce**
This sketch includes all of the settings I use to configure for different LoRa setups, I hope you don't mind. Other than that it is pretty much the Transmit_Interrupt sketch:
``` cpp

#define RADIOLIB_MODULE SX1262  // ESP32 SX1262 (HELTEC)
#define LORA_SS 8
#define LORA_RST 12
#define LORA_DIO 14
#define USE_SX126X

#define FDRS_LORA_TXPWR 17

#define FDRS_LORA_FREQUENCY 915.0
#define FDRS_LORA_BANDWIDTH 125.0
#define FDRS_LORA_SF 7
#define FDRS_LORA_CR 5
#define FDRS_LORA_SYNCWORD 0x12

#include <RadioLib.h>

#ifdef CUSTOM_SPI
#ifdef ESP32
SPIClass SPI1(HSPI);
#endif  // ESP32
RADIOLIB_MODULE radio = new Module(LORA_SS, LORA_DIO, LORA_RST, -1, SPI1);
#else
RADIOLIB_MODULE radio = new Module(LORA_SS, LORA_DIO, LORA_RST, -1);
#endif  // CUSTOM_SPI

void setup() {
  Serial.begin(115200);
  //  delay(5000);
  Serial.println("Hello, world!");

#ifdef CUSTOM_SPI
#ifdef ESP32
  SPI1.begin(LORA_SPI_SCK, LORA_SPI_MISO, LORA_SPI_MOSI);
#endif  // ESP32
#ifdef ARDUINO_ARCH_RP2040
  SPI1.setRX(LORA_SPI_MISO);
  SPI1.setTX(LORA_SPI_MOSI);
  SPI1.setSCK(LORA_SPI_SCK);
  SPI1.begin(false);
#endif  //ARDUINO_ARCH_RP2040
#endif  // CUSTOM_SPI

#ifdef USE_SX126X
  int state = radio.begin(FDRS_LORA_FREQUENCY, FDRS_LORA_BANDWIDTH, FDRS_LORA_SF, FDRS_LORA_CR, FDRS_LORA_SYNCWORD, FDRS_LORA_TXPWR, 8, 1.6, false);
#else
  int state = radio.begin(FDRS_LORA_FREQUENCY, FDRS_LORA_BANDWIDTH, FDRS_LORA_SF, FDRS_LORA_CR, FDRS_LORA_SYNCWORD, FDRS_LORA_TXPWR, 8, 0);
#endif

  if (state == RADIOLIB_ERR_NONE) {
    Serial.println("RadioLib initialization successful!");
  } else {
    Serial.println("RadioLib initialization failed, code " + String(state));
  }


}

void loop() {
  Serial.print(F("[SX1278] Transmitting packet ... "));


  int state = radio.startTransmit("Hello World!");



  if (state == RADIOLIB_ERR_NONE) {
    // the packet was successfully transmitted
    Serial.println(F(" success!"));

    // print measured data rate
    Serial.print(F("[SX1278] Datarate:\t"));
    Serial.print(radio.getDataRate());
    Serial.println(F(" bps"));

  } else if (state == RADIOLIB_ERR_PACKET_TOO_LONG) {
    // the supplied packet was longer than 256 bytes
    Serial.println(F("too long!"));

  } else if (state == RADIOLIB_ERR_TX_TIMEOUT) {
    // timeout occurred while transmitting packet
    Serial.println(F("timeout!"));

  } else {
    // some other error occurred
    Serial.print(F("failed, code "));
    Serial.println(state);

  }

  // wait for a second before transmitting again
  delay(1000);
}

```

**Comment by jgromes** [comment]
A couple observations:

1. You're using the `startTransmit` method, which is only the first half of interrupt-driven transmission. You should also call `finishTransmit` after this is done. Or, just use the blocking `transmit` method.
2. `startTransmit` doesn't measure data rate, only blocking transmission does.

> When this occurs the RSSI drops to 30

I'm assuming the actual value is -30 dBm, which is a lot more than the original -120 dBm. So the signal seems to be several orders of magnitude stronger now.

> do I need to do something with the RF switch?

That depends if your module has an RF switch. If not, then you don't need to control it. If it does, and if it connected to DIO2, then the library can take care of it automatically. If the switch has more control pins (e.g. TXEN and RXEN), then you need to use the `setRfSwitchPins` method.

Next, can you also post the received code and the debug mode output?

**Comment by timmbogner** [comment]
Sure thing. I added finishTransmit and it doesn't change behavior, and also switched to the blocking transmit to no success. I should mention that the actual problem is happening in my larger FDRS project, and I'm doing my best at re-creating the problem on a small scale with these examples. I also only have one SX1262 to test with, so the configs help me test it against SX127X modules. In this post will show the **sender sketch, its serial output with debugging, and then the receiving code.** 
_When debugging is enabled on the sender, the problem still goes away._ 
Sender:
``` cpp

#define RADIOLIB_MODULE SX1262  // ESP32 SX1262 (HELTEC)
#define LORA_SS 8
#define LORA_RST 12
#define LORA_DIO 14
#define USE_SX126X


#define FDRS_LORA_TXPWR 17

#define FDRS_LORA_FREQUENCY 915.0
#define FDRS_LORA_BANDWIDTH 125.0
#define FDRS_LORA_SF 7
#define FDRS_LORA_CR 5
#define FDRS_LORA_SYNCWORD 0x12
#define RADIOLIB_DEBUG
#define RADIOLIB_VERBOSE

#include <RadioLib.h>

#ifdef CUSTOM_SPI
#ifdef ESP32
SPIClass SPI1(HSPI);
#endif  // ESP32
RADIOLIB_MODULE radio = new Module(LORA_SS, LORA_DIO, LORA_RST, -1, SPI1);
#else
RADIOLIB_MODULE radio = new Module(LORA_SS, LORA_DIO, LORA_RST, -1);
#endif  // CUSTOM_SPI

void setup() {
  Serial.begin(115200);
  //  delay(5000);
  Serial.println("Hello, world!");

#ifdef CUSTOM_SPI
#ifdef ESP32
  SPI1.begin(LORA_SPI_SCK, LORA_SPI_MISO, LORA_SPI_MOSI);
#endif  // ESP32
#ifdef ARDUINO_ARCH_RP2040
  SPI1.setRX(LORA_SPI_MISO);
  SPI1.setTX(LORA_SPI_MOSI);
  SPI1.setSCK(LORA_SPI_SCK);
  SPI1.begin(false);
#endif  //ARDUINO_ARCH_RP2040
#endif  // CUSTOM_SPI

#ifdef USE_SX126X
  int state = radio.begin(FDRS_LORA_FREQUENCY, FDRS_LORA_BANDWIDTH, FDRS_LORA_SF, FDRS_LORA_CR, FDRS_LORA_SYNCWORD, FDRS_LORA_TXPWR, 8, 1.6, false);
#else
  int state = radio.begin(FDRS_LORA_FREQUENCY, FDRS_LORA_BANDWIDTH, FDRS_LORA_SF, FDRS_LORA_CR, FDRS_LORA_SYNCWORD, FDRS_LORA_TXPWR, 8, 0);
#endif

  if (state == RADIOLIB_ERR_NONE) {
    Serial.println("RadioLib initialization successful!");
  } else {
    Serial.println("RadioLib initialization failed, code " + String(state));
  }
}

void loop() {
  Serial.print(F("[SX1278] Transmitting packet ... "));
  int state = radio.transmit("Hello World!");
  if (state == RADIOLIB_ERR_NONE) {
    // the packet was successfully transmitted
    Serial.println(F(" success!"));
    // print measured data rate
    Serial.print(F("[SX1278] Datarate:\t"));
    Serial.print(radio.getDataRate());
    Serial.println(F(" bps"));
  } else if (state == RADIOLIB_ERR_PACKET_TOO_LONG) {
    // the supplied packet was longer than 256 bytes
    Serial.println(F("too long!"));
  } else if (state == RADIOLIB_ERR_TX_TIMEOUT) {
    // timeout occurred while transmitting packet
    Serial.println(F("timeout!"));
  } else {
    // some other error occurred
    Serial.print(F("failed, code "));
    Serial.println(state);
  }
  delay(1000);
}
```
#
Output: (The problem goes away when debugging is enabled, so this actually demonstrates correct functioning)
```
09:50:14.199 -> ESP-ROM:esp32s3-20210327
09:50:14.199 -> Build:Mar 27 2021
09:50:14.199 -> rst:0x1 (POWERON),boot:0x8 (SPI_FAST_FLASH_BOOT)
09:50:14.199 -> SPIWP:0xee
09:50:14.199 -> mode:DIO, clock div:1
09:50:14.199 -> load:0x3fce3808,len:0x44c
09:50:14.199 -> load:0x403c9700,len:0xbe4
09:50:14.199 -> load:0x403cc700,len:0x2a38
09:50:14.199 -> entry 0x403c98d4
09:50:14.293 -> Hello, world!
09:50:14.293 -> CMD	80	
09:50:14.293 -> DATW	0	FF	
09:50:14.293 -> 
09:50:14.293 -> CMD	80	
09:50:14.293 -> DATW	0	AA	
09:50:14.293 -> 
09:50:14.293 -> CMD	80	
09:50:14.293 -> DATW	0	A2	
09:50:14.293 -> 
09:50:14.293 -> CMD	C0	
09:50:14.293 -> DATR	0	22	0	22	
09:50:14.293 -> 
09:50:14.293 -> CMD	1D	3	20	
09:50:14.293 -> DATR	0	A2	0	53	0	58	0	31	0	32	0	36	0	31	0	20	0	56	0	32	0	44	0	20	0	32	0	44	0	30	0	32	0	0	
09:50:14.328 -> 
09:50:14.328 -> Found SX126x: RADIOLIB_SX126X_REG_VERSION_STRING:
09:50:14.328 -> 0000320 53 58 31 32 36 31 20 56 32 44 20 32 44 30 32 00 | SX1261 V2D 2D02.
09:50:14.328 -> 
09:50:14.328 -> M	SX126x
09:50:14.328 -> CMD	80	
09:50:14.328 -> DATW	0	0	
09:50:14.328 -> 
09:50:14.328 -> CMD	80	
09:50:14.328 -> DATW	0	AA	
09:50:14.328 -> 
09:50:14.328 -> CMD	80	
09:50:14.328 -> DATW	0	A2	
09:50:14.365 -> 
09:50:14.365 -> CMD	C0	
09:50:14.365 -> DATR	0	22	0	22	
09:50:14.365 -> 
09:50:14.365 -> CMD	80	
09:50:14.365 -> DATW	0	A2	
09:50:14.365 -> 
09:50:14.365 -> CMD	C0	
09:50:14.365 -> DATR	0	22	0	22	
09:50:14.365 -> 
09:50:14.365 -> CMD	80	
09:50:14.365 -> DATW	0	A2	
09:50:14.365 -> 
09:50:14.365 -> CMD	C0	
09:50:14.365 -> DATR	0	22	0	22	
09:50:14.365 -> 
09:50:14.365 -> CMD	17	
09:50:14.365 -> DATR	0	A2	0	0	0	20	
09:50:14.365 -> 
09:50:14.365 -> CMD	C0	
09:50:14.365 -> DATR	0	22	0	22	
09:50:14.365 -> 
09:50:14.365 -> CMD	7	
09:50:14.365 -> DATW	0	A2	0	A2	
09:50:14.365 -> 
09:50:14.365 -> CMD	C0	
09:50:14.365 -> DATR	0	22	0	22	
09:50:14.365 -> 
09:50:14.365 -> CMD	97	
09:50:14.365 -> DATW	0	A2	0	A2	1	A2	40	A2	
09:50:14.365 -> 
09:50:14.365 -> CMD	C0	
09:50:14.365 -> DATR	0	22	0	22	
09:50:14.365 -> 
09:50:14.365 -> CMD	8F	
09:50:14.365 -> DATW	0	A2	0	A2	
09:50:14.365 -> 
09:50:14.365 -> CMD	C0	
09:50:14.365 -> DATR	0	22	0	22	
09:50:14.365 -> 
09:50:14.365 -> CMD	8A	
09:50:14.365 -> DATW	1	A2	
09:50:14.365 -> 
09:50:14.365 -> CMD	C0	
09:50:14.365 -> DATR	0	22	0	22	
09:50:14.365 -> 
09:50:14.365 -> CMD	93	
09:50:14.401 -> DATW	20	A2	
09:50:14.401 -> 
09:50:14.401 -> CMD	C0	
09:50:14.401 -> DATR	0	22	0	22	
09:50:14.401 -> 
09:50:14.401 -> CMD	88	
09:50:14.401 -> DATW	3	A2	16	A2	A	A2	0	A2	0	A2	0	A2	0	A2	
09:50:14.401 -> 
09:50:14.401 -> CMD	C0	
09:50:14.401 -> DATR	0	22	0	22	
09:50:14.401 -> 
09:50:14.401 -> CMD	2	
09:50:14.401 -> DATW	43	A2	FF	A2	
09:50:14.401 -> 
09:50:14.401 -> CMD	C0	
09:50:14.401 -> DATR	0	22	0	22	
09:50:14.401 -> 
09:50:14.401 -> CMD	8	
09:50:14.401 -> DATW	0	A2	0	A2	0	A2	0	A2	0	A2	0	A2	0	A2	0	A2	
09:50:14.401 -> 
09:50:14.401 -> CMD	C0	
09:50:14.401 -> DATR	0	22	0	22	
09:50:14.401 -> 
09:50:14.401 -> CMD	89	
09:50:14.401 -> DATW	7F	A2	
09:50:14.401 -> 
09:50:14.401 -> CMD	C0	
09:50:14.401 -> DATR	0	22	0	22	
09:50:14.401 -> 
09:50:14.401 -> CMD	11	
09:50:14.401 -> DATR	0	B2	0	1	
09:50:14.401 -> 
09:50:14.401 -> CMD	C0	
09:50:14.401 -> DATR	0	B2	0	32	
09:50:14.401 -> 
09:50:14.401 -> Symbol length: 1.02 ms
09:50:14.401 -> CMD	8B	
09:50:14.436 -> DATW	9	C2	6	C2	1	C2	0	C2	
09:50:14.436 -> 
09:50:14.436 -> CMD	C0	
09:50:14.436 -> DATR	0	22	0	22	
09:50:14.436 -> 
09:50:14.436 -> CMD	11	
09:50:14.436 -> DATR	0	A2	0	1	
09:50:14.436 -> 
09:50:14.436 -> CMD	C0	
09:50:14.436 -> DATR	0	22	0	22	
09:50:14.436 -> 
09:50:14.436 -> CMD	D	7	40	
09:50:14.436 -> DATW	14	A2	24	A2	
09:50:14.436 -> 
09:50:14.436 -> CMD	11	
09:50:14.436 -> DATR	0	A2	0	1	
09:50:14.436 -> 
09:50:14.436 -> CMD	C0	
09:50:14.436 -> DATR	0	22	0	22	
09:50:14.436 -> 
09:50:14.436 -> CMD	1D	7	36	
09:50:14.436 -> DATR	0	A2	0	D	
09:50:14.436 -> 
09:50:14.436 -> CMD	C0	
09:50:14.436 -> DATR	0	22	0	22	
09:50:14.436 -> 
09:50:14.436 -> CMD	D	7	36	
09:50:14.436 -> DATW	D	A2	
09:50:14.436 -> 
09:50:14.436 -> CMD	8C	
09:50:14.436 -> DATW	0	A2	8	A2	0	A2	FF	A2	1	A2	0	A2	
09:50:14.436 -> 
09:50:14.436 -> CMD	C0	
09:50:14.436 -> DATR	0	22	0	22	
09:50:14.436 -> 
09:50:14.436 -> CMD	D	8	E7	
09:50:14.436 -> DATW	18	A2	
09:50:14.436 -> 
09:50:14.436 -> CMD	9D	
09:50:14.436 -> DATW	1	A2	
09:50:14.436 -> 
09:50:14.472 -> CMD	C0	
09:50:14.472 -> DATR	0	22	0	22	
09:50:14.472 -> 
09:50:14.472 -> CMD	96	
09:50:14.472 -> DATW	1	A2	
09:50:14.472 -> 
09:50:14.472 -> CMD	C0	
09:50:14.472 -> DATR	0	22	0	22	
09:50:14.472 -> 
09:50:14.472 -> CMD	11	
09:50:14.472 -> DATR	0	A2	0	1	
09:50:14.472 -> 
09:50:14.472 -> CMD	C0	
09:50:14.472 -> DATR	0	22	0	22	
09:50:14.472 -> 
09:50:14.472 -> Symbol length: 0.26 ms
09:50:14.472 -> CMD	8B	
09:50:14.472 -> DATW	7	A2	6	A2	1	A2	0	A2	
09:50:14.472 -> 
09:50:14.472 -> CMD	C0	
09:50:14.472 -> DATR	0	22	0	22	
09:50:14.472 -> 
09:50:14.472 -> CMD	11	
09:50:14.472 -> DATR	0	A2	0	1	
09:50:14.472 -> 
09:50:14.472 -> CMD	C0	
09:50:14.472 -> DATR	0	22	0	22	
09:50:14.472 -> 
09:50:14.472 -> Symbol length: 1.02 ms
09:50:14.472 -> CMD	8B	
09:50:14.472 -> DATW	7	A2	4	A2	1	A2	0	A2	
09:50:14.472 -> 
09:50:14.472 -> CMD	C0	
09:50:14.472 -> DATR	0	22	0	22	
09:50:14.472 -> 
09:50:14.472 -> CMD	98	
09:50:14.472 -> DATW	E1	A2	E9	A2	
09:50:14.472 -> 
09:50:14.472 -> CMD	C0	
09:50:14.513 -> DATR	0	22	0	22	
09:50:14.513 -> 
09:50:14.513 -> CMD	86	
09:50:14.513 -> DATW	39	A2	30	A2	0	A2	0	A2	
09:50:14.513 -> 
09:50:14.513 -> CMD	C0	
09:50:14.513 -> DATR	0	22	0	22	
09:50:14.513 -> 
09:50:14.513 -> CMD	1D	8	D8	
09:50:14.513 -> DATR	0	A2	0	C8	
09:50:14.513 -> 
09:50:14.513 -> CMD	C0	
09:50:14.513 -> DATR	0	22	0	22	
09:50:14.513 -> 
09:50:14.513 -> CMD	D	8	D8	
09:50:14.513 -> DATW	DE	A2	
09:50:14.513 -> 
09:50:14.513 -> CMD	1D	8	E7	
09:50:14.513 -> DATR	0	A2	0	18	
09:50:14.513 -> 
09:50:14.513 -> CMD	C0	
09:50:14.513 -> DATR	0	22	0	22	
09:50:14.513 -> 
09:50:14.513 -> CMD	95	
09:50:14.513 -> DATW	4	A2	7	A2	0	A2	1	A2	
09:50:14.513 -> 
09:50:14.513 -> CMD	C0	
09:50:14.513 -> DATR	0	22	0	22	
09:50:14.513 -> 
09:50:14.513 -> CMD	8E	
09:50:14.513 -> DATW	11	A2	4	A2	
09:50:14.513 -> 
09:50:14.513 -> CMD	C0	
09:50:14.513 -> DATR	0	22	0	22	
09:50:14.513 -> 
09:50:14.513 -> CMD	D	8	E7	
09:50:14.513 -> DATW	18	A2	
09:50:14.513 -> 
09:50:14.513 -> RadioLib initialization successful!
09:50:14.513 -> [SX1278] Transmitting packet ... CMD	80	
09:50:14.550 -> DATW	0	A2	
09:50:14.550 -> 
09:50:14.550 -> CMD	C0	
09:50:14.550 -> DATR	0	22	0	22	
09:50:14.550 -> 
09:50:14.550 -> CMD	11	
09:50:14.550 -> DATR	0	A2	0	1	
09:50:14.550 -> 
09:50:14.550 -> CMD	C0	
09:50:14.550 -> DATR	0	22	0	22	
09:50:14.550 -> 
09:50:14.550 -> CMD	11	
09:50:14.550 -> DATR	0	A2	0	1	
09:50:14.550 -> 
09:50:14.550 -> CMD	C0	
09:50:14.550 -> DATR	0	22	0	22	
09:50:14.550 -> 
09:50:14.550 -> Timeout in 61824 us
09:50:14.550 -> CMD	11	
09:50:14.550 -> DATR	0	A2	0	1	
09:50:14.550 -> 
09:50:14.550 -> CMD	C0	
09:50:14.550 -> DATR	0	22	0	22	
09:50:14.550 -> 
09:50:14.550 -> CMD	1D	7	36	
09:50:14.550 -> DATR	0	A2	0	D	
09:50:14.550 -> 
09:50:14.550 -> CMD	C0	
09:50:14.550 -> DATR	0	22	0	22	
09:50:14.550 -> 
09:50:14.550 -> CMD	D	7	36	
09:50:14.550 -> DATW	D	A2	
09:50:14.550 -> 
09:50:14.550 -> CMD	8C	
09:50:14.550 -> DATW	0	A2	8	A2	0	A2	C	A2	1	A2	0	A2	
09:50:14.550 -> 
09:50:14.550 -> CMD	C0	
09:50:14.550 -> DATR	0	22	0	22	
09:50:14.550 -> 
09:50:14.550 -> CMD	8	
09:50:14.550 -> DATW	2	A2	1	A2	0	A2	1	A2	0	A2	0	A2	0	A2	0	A2	
09:50:14.597 -> 
09:50:14.597 -> CMD	C0	
09:50:14.597 -> DATR	0	22	0	22	
09:50:14.597 -> 
09:50:14.597 -> CMD	8F	
09:50:14.597 -> DATW	0	A2	0	A2	
09:50:14.597 -> 
09:50:14.597 -> CMD	C0	
09:50:14.597 -> DATR	0	22	0	22	
09:50:14.597 -> 
09:50:14.597 -> CMD	E	0	
09:50:14.597 -> DATW	48	A2	65	A2	6C	A2	6C	A2	6F	A2	20	A2	57	A2	6F	A2	72	A2	6C	A2	64	A2	21	A2	
09:50:14.597 -> 
09:50:14.597 -> CMD	C0	
09:50:14.597 -> DATR	0	22	0	22	
09:50:14.597 -> 
09:50:14.597 -> CMD	2	
09:50:14.597 -> DATW	43	A2	FF	A2	
09:50:14.597 -> 
09:50:14.597 -> CMD	C0	
09:50:14.597 -> DATR	0	22	0	22	
09:50:14.597 -> 
09:50:14.597 -> CMD	1D	8	89	
09:50:14.597 -> DATR	0	A2	0	4	
09:50:14.597 -> 
09:50:14.597 -> CMD	C0	
09:50:14.597 -> DATR	0	22	0	22	
09:50:14.597 -> 
09:50:14.597 -> CMD	11	
09:50:14.597 -> DATR	0	A2	0	1	
09:50:14.597 -> 
09:50:14.597 -> CMD	C0	
09:50:14.597 -> DATR	0	22	0	22	
09:50:14.597 -> 
09:50:14.597 -> CMD	D	8	89	
09:50:14.597 -> DATW	4	A2	
09:50:14.597 -> 
09:50:14.597 -> CMD	83	
09:50:14.597 -> DATW	0	A2	0	A2	0	A2	
09:50:14.597 -> 
09:50:14.597 -> CMD	C0	
09:50:14.597 -> DATR	0	22	0	22	
09:50:14.597 -> 
09:50:14.635 -> CMD	2	
09:50:14.635 -> DATW	43	AC	FF	AC	
09:50:14.635 -> 
09:50:14.635 -> CMD	C0	
09:50:14.635 -> DATR	0	2C	0	2C	
09:50:14.635 -> 
09:50:14.635 -> CMD	80	
09:50:14.635 -> DATW	0	AC	
09:50:14.687 -> 
09:50:14.687 -> CMD	C0	
09:50:14.687 -> DATR	0	22	0	22	
09:50:14.687 -> 
09:50:14.687 ->  success!
09:50:14.687 -> [SX1278] Datarate:	2072.49 bps
```

#
Receiver:
``` cpp

#define RADIOLIB_MODULE SX1276  // ESP32 SX1276 (TTGO)
#define LORA_SS 18
#define LORA_RST 14
#define LORA_DIO 26

#define FDRS_LORA_TXPWR 17

#define FDRS_LORA_FREQUENCY 915.0
#define FDRS_LORA_BANDWIDTH 125.0
#define FDRS_LORA_SF 7
#define FDRS_LORA_CR 5
#define FDRS_LORA_SYNCWORD 0x12

#include <RadioLib.h>

#ifdef CUSTOM_SPI
#ifdef ESP32
SPIClass SPI1(HSPI);
#endif  // ESP32
RADIOLIB_MODULE radio = new Module(LORA_SS, LORA_DIO, LORA_RST, -1, SPI1);
#else
RADIOLIB_MODULE radio = new Module(LORA_SS, LORA_DIO, LORA_RST, -1);
#endif  // CUSTOM_SPI

void setup() {
  Serial.begin(115200);
  //  delay(5000);
  Serial.println("Hello, world!");

#ifdef CUSTOM_SPI
#ifdef ESP32
  SPI1.begin(LORA_SPI_SCK, LORA_SPI_MISO, LORA_SPI_MOSI);
#endif  // ESP32
#ifdef ARDUINO_ARCH_RP2040
  SPI1.setRX(LORA_SPI_MISO);
  SPI1.setTX(LORA_SPI_MOSI);
  SPI1.setSCK(LORA_SPI_SCK);
  SPI1.begin(false);
#endif  //ARDUINO_ARCH_RP2040
#endif  // CUSTOM_SPI

#ifdef USE_SX126X
  int state = radio.begin(FDRS_LORA_FREQUENCY, FDRS_LORA_BANDWIDTH, FDRS_LORA_SF, FDRS_LORA_CR, FDRS_LORA_SYNCWORD, FDRS_LORA_TXPWR, 8, 1.6, false);
#else
  int state = radio.begin(FDRS_LORA_FREQUENCY, FDRS_LORA_BANDWIDTH, FDRS_LORA_SF, FDRS_LORA_CR, FDRS_LORA_SYNCWORD, FDRS_LORA_TXPWR, 8, 0);
#endif

  if (state == RADIOLIB_ERR_NONE) {
    Serial.println("RadioLib initialization successful!");
  } else {
    Serial.println("RadioLib initialization failed, code " + String(state));
  }
#ifdef USE_SX126X
  radio.setDio1Action(setFlag);
#else
  radio.setDio0Action(setFlag);
#endif

  // start listening for LoRa packets
  Serial.print(F("[SX1278] Starting to listen ... "));
  state = radio.startReceive();
  if (state == RADIOLIB_ERR_NONE) {
    Serial.println(F("success!"));
  } else {
    Serial.print(F("failed, code "));
    Serial.println(state);
    while (true)
      ;
  }
}

// flag to indicate that a packet was received
volatile bool receivedFlag = false;

#if defined(ESP8266) || defined(ESP32)
ICACHE_RAM_ATTR
#endif
void setFlag(void) {
  receivedFlag = true;
}


void loop() {
  // check if the flag is set
  if (receivedFlag) {
    // reset flag
    receivedFlag = false;

    // you can read received data as an Arduino String
    String str;
    int state = radio.readData(str);

    if (state == RADIOLIB_ERR_NONE) {
      // packet was successfully received
      Serial.println(F("[SX1278] Received packet!"));

      // print data of the packet
      Serial.print(F("[SX1278] Data:\t\t"));
      Serial.println(str);

      // print RSSI (Received Signal Strength Indicator)
      Serial.print(F("[SX1278] RSSI:\t\t"));
      Serial.print(radio.getRSSI());
      Serial.println(F(" dBm"));

      // print SNR (Signal-to-Noise Ratio)
      Serial.print(F("[SX1278] SNR:\t\t"));
      Serial.print(radio.getSNR());
      Serial.println(F(" dB"));

    } else if (state == RADIOLIB_ERR_CRC_MISMATCH) {
      // packet was received, but is malformed
      Serial.println(F("[SX1278] CRC error!"));

    } else {
      // some other error occurred
      Serial.print(F("[SX1278] Failed, code "));
      Serial.println(state);
    }

    // put module back to listen mode
    radio.startReceive();
  }
}
```

**Comment by jgromes** [comment]
TL;DR: On SX1262, you're missing the BUSY pin.

Long version:

For a while, I was unable to reproduce this. I even pulled out some breadboards to replicate the setup as close as I could, with ESP32/SX1262 transmitter, and ESP32/SX1278 receiver. I was unable to get it to break with my default configuration, so I started replicating yours. Even setting the same frequency etc. had no effect, but then I noticed this line in your examples:

```c++
RADIOLIB_MODULE radio = new Module(LORA_SS, LORA_DIO, LORA_RST, -1);
```

In all cases, you are setting the last pin of the Module constructor to -1 (BTW the correct RadioLib alias for unconnected pin is `RADIOLIB_NC`). For SX127x, this is fine as the second DIO pin (DIO1 in this case) is optional and only needed for some functionality.

However, in the case of SX126x, the last pin is BUSY, and it's not optional. It lets RadioLib know when the chip is ... well, busy. Since this signal is used to control timing of SPI transactions  as well as calibrations, setting it to -1 (which prevents RadioLib from using it) will introduce timing issues: for example, RadioLib might try to initiate an SPI transaction before the previous one was executed.

However, in debug+verbose, there's the additional overhead of printing everything to Serial. It seems like this slowdown is sufficient for the SPI transactions to get executed.

Why wasn't this broken pre-5.7.0, I'm not sure. There was a slight rework of the SPI for SX126x in 5.7.0, so it is possible there was an unrelated issue that was having a similar effect without debug.

PS: It looks like SX1262 modules will always report as SX1261.

**Comment by jgromes** [comment]
No worries, I'm glad you were able to get it working!
